e2e-tests/production-runtime/cypress/integration/lifecycle-methods.js

```diff
@@ -0,0 +1,41 @@
+describe(`Production build tests`, () => {
+  it(`should remount when navigating to different template`, () => {
+    cy.visit(`/`).waitForAPI(`onRouteUpdate`)
+
+    cy.getTestElement(`page2`)
+      .click()
+      .waitForAPI(`onRouteUpdate`)
+
+    cy.window().then(win => {
+      cy.wrap(win.___PageComponentLifecycleCallsLog).snapshot()
```

Comment:
using snapshot testing, which is quite verbose and might be hard to read - it might make sense to not use snapshots here and use more programmatic assertions

Reply:
One way to test this would be count renders in local state and add that to the DOM and assert against it